src/System.Linq.Expressions/src/System/Runtime/CompilerServices/ReadOnlyCollectionBuilder.cs

```diff
@@ -131,8 +163,13 @@ public int IndexOf(T item)
         /// <param name="item">The object to insert into the <see cref="ReadOnlyCollectionBuilder{T}"/>.</param>
         public void Insert(int index, T item)
         {
-            if (index > _size)
+            // Note that insertions at the end are legal.
+            // Following trick can reduce the range check by one
```

Comment:
Then say that in the comment (though it's one comparison *and* one branch as `||` causes a branch). I don't think this needs a comment at all, though that's a much more subjective matter, but if it's in then "by one comparison" is clearer than "by one"